pkg/microservice/aslan/core/environment/service/environment.go

```diff
@@ -2659,7 +2742,36 @@ func getUpdatedProductServices(updateProduct *commonmodels.Product, serviceRevis
 	return updatedAllServices
 }
 
-func updateProductGroup(productName, envName, updateType string, productResp *commonmodels.Product, currentProductServices [][]*commonmodels.ProductService, overrideCharts []*commonservice.RenderChartArg, log *zap.SugaredLogger) error {
+func intervalExecutor(interval time.Duration, serviceList []interface{}, handler func(data interface{}, log *zap.SugaredLogger) error, log *zap.SugaredLogger) *multierror.Error {
```

Comment:
*multierror.Error => error
if you want to convert it back to a multierror.Error, just do it in upper functions.